lib/injectable/class_methods.rb

```diff
@@ -153,5 +180,26 @@ def required_call_arguments
     def find_required_arguments(hash)
       hash.reject { |_arg, options| options.key?(:default) }.keys
     end
+
+    def initialize_single_return(type)
+      raise(ArgumentError, ':type for returns must be a Class or Module') unless type.is_a?(Module)
+
+      { kind: :single, type: type }
+    end
+
+    def initialize_collection_return(collection_class, element_class)
+      raise(ArgumentError, ':of for returns must be a Class or Module') unless element_class.is_a?(Module)
+
+      valid_collection = collection_class.is_a?(Class) || collection_class.is_a?(Module)
```

Comment:
Confusion: should I validate for Class and Module? Just for Class, just for Module?